third_party/move/move-model/src/well_known.rs

```diff
@@ -45,15 +45,21 @@ pub const TYPE_NAME_GET_SPEC: &str = "type_name::$get";
 /// The well-known name of the first parameter of a method.
 pub const RECEIVER_PARAM_NAME: &str = "self";
 
-/// The well-known abort codes used by the compiler. These are marked
-/// by lowest 6 bytes of a sha256 of the string "Move 2 Abort Code",
-/// appended with two bytes for the error type.
-/// TODO: add a check at runtime that user is not clashing with reserved
-/// codes?
-pub const WELL_KNOWN_ABORT_CODE_BASE: u64 = 0xD8CA26CBD9BE << 16;
+/// The well-known abort codes used by the compiler. Those confirm
+/// to the error category standard as defined in `errors.move`
+/// in the standard library. The lowest three bytes represent
+/// the error category (one byte) and the reason (two bytes).
+/// All compiler generated abort cords use category
+/// `std::error::INTERNAL` (`0xB`). The upper five bytes
+/// are populated with the lowest bytes of the sha256
+/// of the string "Move 2 Abort Code".
+const fn make_abort_code(n: u64) -> u64 {
```

Comment:
```suggestion
const fn make_abort_code(reason: u16) -> u64 {
```

From what I can tell from the comment above, `n` is reason, and is two bytes, so we can use a stricter type.